src/composed/signed_key/secret.rs

```diff
@@ -234,8 +234,8 @@ impl PublicKeyTrait for SignedSecretKey {
         self.primary_key.verify_signature(hash, data, sig)
     }
 
-    fn encrypt<R: Rng + CryptoRng>(&self, rng: R, plain: &[u8]) -> Result<Vec<Mpi>> {
-        self.primary_key.encrypt(rng, plain)
+    fn encrypt<R: Rng + CryptoRng>(&self, rng: R, plain: &[u8], v6_esk: bool) -> Result<EskBytes> {
```

Comment:
this is very unfortunte

Reply:
I made this slightly less horrible with an enum: 6cc62b1d00afc358dd26fe2094b2ff4acab92bbe ... but if we come up with better ideas, I'm all into iterating further.